### DISCUSSION TITLE
How to integrate pusher and laravel echo with livewire ?

### DISCUSSION THREAD

**Discussion by 0xayman** [livewire/livewire#4462] [1. Help]
I'm trying to use puhser & laravel echo to make realtime notification system in a livewire app. I installed all the required libraries. but when I try to register any listener on any livewire componnet I got a warning on the browser console saying that `Laravel Echo cannot be found`.

I've configured Laravel Echo & Pusher on `bootstrap.js` file:
```
import Echo from 'laravel-echo'

window.Pusher = require('pusher-js');

window.Echo = new Echo({
    broadcaster: 'pusher',
    key: process.env.MIX_PUSHER_APP_KEY,
    cluster: process.env.MIX_PUSHER_APP_CLUSTER,
    forceTLS: true
});
```

Then I tried to listen to a channel on my component:
```
protected $listeners = [
        "echo:notificationsChannel"  => 'notifyUser'
    ];
``` 

My `app.blade.php` file:
```
<!DOCTYPE html>
<html class="dark" lang="{{ str_replace('_', '-', app()->getLocale()) }}">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    @livewireStyles
</head>

<body class="bg-white dark:bg-gray-900">

    // some content

    @livewireScripts

    <script src="{{ mix('js/app.js') }}"></script>
</body>

</html>
```

**Comment by joshhanley** [comment]
@0xayman try moving `@livewireScripts` to after your `js/app.js` script and see if that works.

Hope this helps!
